zowe/zowe-cli
# feat(secrets): Use `core-foundation-rs` instead of `security-framework` for macOS logic

**What It Does**

- Updated macOS logic to use the `core-foundation-rs` crate and FFI calls instead of the `security-framework` crate, as `security-framework` has been archived.
- Updates Zowe CLI to use `7.18.6` of the Secrets SDK that contains the above changes.
- Fixes a bug with the Linux implementation: removed unnecessary `libsecret::secret_value_unref` call when interacting with items (causes segfault in scenarios where only one match is found)

**Note:** Once the required # of reviews is met, I'm going to publish Secrets SDK 7.18.6 before merging.

**How to Test**

Try to build, install, and use Zowe CLI using this PR on macOS. The CLI should work as expected using secure credentials.

If you want to test the Linux fix for edge-cases with the `findCredentials` function, install the Secrets SDK as a global package, then save the following script as a `.js` file and execute it with Node (`node test.js`):

```js
const { keyring } = require("@zowe/secrets-for-zowe-sdk");
await keyring.setPassword("TestService", "TestOneCred", "pass");
await keyring.findCredentials("TestService");
console.log("this line should be printed!");
await keyring.deletePassword("TestService", "TestOneCred");
```
The script should run without any errors or segmentation faults printed to stdout.

**Review Checklist**
I certify that I have:
